src/main/java/com/julienviet/pgclient/Numeric.java

```
@@ -0,0 +1,162 @@
+   *
+   * @param number the number
+   * @return the {@code Numeric} value
+   * @throws NumberFormatException when the number is infinite
+   */
+  public static Numeric create(Number number) {
+    if (number == null) {
+      throw new NullPointerException();
+    }
+    if (number instanceof Double && ((Double)number).isInfinite() || number instanceof Float && ((Float)number).isInfinite()) {
+      throw new NumberFormatException("Infinite numbers are not valid numerics");
+    }
+    return new Numeric(number);
+  }
+
+  /**
+   * Parse and return a {@code Numeric} instance for the given {@code s}.
+   * <p/>
+   * The string {@code "Nan"} will return the {@link #NaN} instance.
+   *
+   * @param s the string
+   * @return the {@code Numeric} value
+   */
+  public static Numeric parse(String s) {
+    switch (s) {
+      case "NaN":
+        return NaN;
+      default:
+        return new Numeric(new BigDecimal(s));
+    }
+  }
+
+  private Numeric(Number value) {
+    this.value = value;
+  }
+
+  @Override
+  public int intValue() {
+    return value.intValue();
+  }
+
+  @Override
+  public long longValue() {
+    return value.longValue();
+  }
+
+  @Override
+  public float floatValue() {
+    return value.floatValue();
+  }
+
+  @Override
+  public double doubleValue() {
+    return value.doubleValue();
+  }
+
+  /**
+   * @return {@code true} when this number represents {@code NaN}
+   */
+  public boolean isNaN() {
+    return value instanceof Double && ((Double)value).isNaN() || value instanceof Float && ((Float)value).isNaN();
+  }
+
+  /**
+   * @return  the numeric value represented by this object after conversion
+   *          to type {@code BigDecimal}.
+   */
+  public BigDecimal bigDecimalValue() {
+    if (value instanceof BigDecimal) {
+      return (BigDecimal) value;
+    } else if (value instanceof BigInteger) {
+      return new BigDecimal((BigInteger)value);
+    } else if (isNaN()) {
```

Perhaps rather throw an exception in that case? Or document it at least, as getting null may be surprising otherwise.

---

I think it should be documented, that's a situation that the developer can also take in account using `isNaN()` method before.